docs/docs/data-fetching.md

```diff
@@ -0,0 +1,169 @@
+
+### Fetching data at build time
+
+For fetching data at build time, you can create an integration with a third-party system by sourcing data in your `gatsby-node` file and creating nodes for the GraphQL layer that become queryable in pages. This is the same method that source plugins implement to [source data](/docs/content-and-data/) while the site builds. You can read about that process in the [Creating a Source Plugin guide](/docs/creating-a-source-plugin/).
+
+> This process of fetching data at build time and creating pages from the data is [covered in more depth in the tutorial](/tutorial/part-five/) as well as the docs for [creatiing pages from data programmatically](/docs/programmatically-create-pages-from-data/).
+
+#### Source data to be queried at build time
+
+The simplest way to source data is to take advantage of an existing source plugin. To install the `gatsby-source-graphql`:
+
+```shell
+npm install --save gatsby-source-graphql
+```
+
+Then, add the plugin to your `gatsby-config`:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-source-graphql`,
+      options: {
+        typeName: `GitHub`,
+        fieldName: `github`,
+        url: `https://api.github.com/graphql`,
+        headers: {
+          Authorization: `Bearer your-github-token`,
+        },
+      },
+    },
+  ],
+}
+```
+
+> Because the GitHub GraphQL API requires you to be authenticated to make requests, you need to create a token that you would replace in the header for Authorization that says "your-github-token". You can [secure your key using environment variables](/docs/environment-variables/) if you're pushing code to a public repository.
+
+Source plugins take advantage of the [`sourceNodes` API](/docs/node-apis/#sourceNodes) and the [`createNode` action](/docs/actions/#createNode) provided by Gatsby to make your data queryable during the build process.
+
+#### Writing a query to gather the static data needed for a page
+
+With the source plugin installed and added to your config, you can write a [static query](/docs/use-static-query/) that will pull the data from GraphQL API and will retrieve the necessary data while building the site.
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { graphql, useStaticQuery } from "gatsby" // highlight-line
+
+const IndexPage = () => {
+  // highlight-start
+  const gatsbyRepoData = useStaticQuery(graphql`
+    query {
+      github {
+        repository(name: "gatsby", owner: "gatsbyjs") {
+          id
+          nameWithOwner
+          url
+        }
+      }
+    }
+  `)
+  // highlight-end
+
+  return (
+    <section>
+      <p>
+        Build Time Data: Gatsby repo{` `}
+        <a href="gatsbyRepoData.github.repository.nameWithOwner.url">
+          {gatsbyRepoData.github.repository.nameWithOwner} // highlight-line
+        </a>
+      </p>
+    </section>
+  )
+}
+
+export default IndexPage
+```
+
```

Comment:
Perhaps there should be a build-time example rendered on page to be consistent with the client-side example?